File: apps/api/app/models/project.py
from pydantic import BaseModel, ConfigDict, Field


class Project(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    id: str
    title: str
    source_language: str = Field(alias="sourceLanguage")
    image_dir: str = Field(alias="imageDir")
    created_at: str = Field(alias="createdAt")
    updated_at: str = Field(alias="updatedAt")

File: packages/schema/src/project.ts
import { z } from "zod";

export const projectSchema = z.object({
  id: z.string().min(1),
  title: z.string().min(1),
  sourceLanguage: z.enum(["ja", "zh", "en"]),
  imageDir: z.string().min(1),
  createdAt: z.string().datetime(),
  updatedAt: z.string().datetime(),
});

export type Project = z.infer<typeof projectSchema>;
